internal/pkg/consensus/expected_test.go

```diff
@@ -72,57 +73,20 @@ func TestExpected_RunStateTransition_validateMining(t *testing.T) {
+		exp := consensus.NewExpected(cistore, bstore, consensus.NewDefaultProcessor(&consensus.FakeChainRandomness{}), &views, th.BlockTimeTest,
+			&consensus.FailingElectionValidator{}, &consensus.FakeTicketMachine{}, &proofs.ElectionPoster{})
 
 		nextBlocks := requireMakeNBlocks(t, 3, pTipSet, genesisBlock.StateRoot.Cid, types.EmptyReceiptsCID, miners, minerToWorker, mockSigner)
 		tipSet := th.RequireNewTipSet(t, nextBlocks...)
 
 		emptyBLSMessages, emptyMessages := emptyMessages(len(nextBlocks))
 
-		_, _, err = exp.RunStateTransition(ctx, tipSet, emptyBLSMessages, emptyMessages, []block.TipSet{pTipSet}, nextBlocks[0].ParentWeight, genesisBlock.StateRoot.Cid, genesisBlock.MessageReceipts.Cid)
+		_, _, err = exp.RunStateTransition(ctx, tipSet, emptyBLSMessages, emptyMessages, genesisBlock.ParentWeight, genesisBlock.StateRoot.Cid, genesisBlock.MessageReceipts.Cid)
 		require.Error(t, err)
 		assert.True(t, strings.Contains(err.Error(), "lost election"))
 	})
 
-	t.Run("correct tickets processed in election and next ticket", func(t *testing.T) {
-		cistore, bstore := setupCborBlockstore()
-		genesisBlock, err := th.DefaultGenesis(cistore, bstore)
-		require.NoError(t, err)
-
-		pTipSet := th.RequireNewTipSet(t, genesisBlock)
-
-		nextRoot, miners, m2w := setTree(ctx, t, kis, cistore, bstore, genesisBlock.StateRoot.Cid)
-		views := consensus.AsPowerStateViewer(appstate.NewViewer(cistore))
-		ancestors := make([]block.TipSet, 5)
-		for i := 0; i < int(miner.ElectionLookback); i++ {
-			ancestorBlk := requireMakeNBlocks(t, 1, pTipSet, nextRoot, types.EmptyReceiptsCID, miners, m2w, mockSigner)
-			ancestors[i] = th.RequireNewTipSet(t, ancestorBlk...)
-			pTipSet = ancestors[i]
-		}
-
-		isLookingBack := func(ticket block.Ticket) {
-			expTicket, err := ancestors[miner.ElectionLookback-1].MinTicket()
-			require.NoError(t, err)
-			assert.Equal(t, expTicket, ticket)
-		}
-		mockElection := consensus.NewMockElectionMachine(isLookingBack)
-
-		isOneBack := func(ticket block.Ticket) {
-			expTicket, err := ancestors[0].MinTicket()
-			require.NoError(t, err)
-			assert.Equal(t, expTicket, ticket)
-		}
-		mockTicketGen := consensus.NewMockTicketMachine(isOneBack)
-
-		exp := consensus.NewExpected(cistore, bstore, th.NewFakeProcessor(), &views, th.BlockTimeTest, mockElection, mockTicketGen, &proofs.ElectionPoster{})
-
-		nextBlocks := requireMakeNBlocks(t, 3, pTipSet, nextRoot, types.EmptyReceiptsCID, miners, m2w, mockSigner)
-		tipSet := th.RequireNewTipSet(t, nextBlocks...)
-
-		emptyBLSMessages, emptyMessages := emptyMessages(len(nextBlocks))
-		_, _, err = exp.RunStateTransition(ctx, tipSet, emptyBLSMessages, emptyMessages, ancestors, nextBlocks[0].ParentWeight, nextRoot, nextBlocks[0].MessageReceipts.Cid)
-		assert.NoError(t, err)
-	})
+	// TODO: test that the correct tickets are processed for election and ticket generation
```

Comment:
This test (and use of the mock ticket/election machines) was too hard to keep since tickets are no longer simply generated from each other.